### PR TITLE
Change Playbook <=> Gear Association to Many to Many

### LINKED ISSUE #96: Convert Gear <=> Playbook relation to a many-to-many
## Describe the Feature
**Is your feature request related to a problem? Please describe.**
Right now gear `belongs_to` playbook. But many playbooks have access to the same weapons and those weapons have the same tags.

**Additional context**

## Describe the Technical Implementation of the Solution
**Solution**
- Create a join table for playbooks/gear
- Move all duplicate gear to the `gear.seeds.rb` file
- reference gear/playbook in playbooks/<playbook>.seeds.rb
- Update hunter show/edit to include new reference for gear

**References**
Any references to materials or help articles that may help in implementing this issue.

### PR DESCRIPTION
**Pull request by LeftoverCurry**
### **Issue:**
Gear was originally set up to belong to an individual playbook, but in reality the same gear could belong to many different playbooks.

### **Resolution:**
* Created a `has_many: :through` ActiveRecord association for each table and set up a through table with the model PlaybookGear.  
* Removed `playbook_id` from the Gear table.
* Restructured the currently built out Playbook seeds to no longer reference the gear.
* Restructured and built out the Gear seeds to contain a static list of available gear and their respective playbooks.
* Created a new tag type called 'keywords' that can distinguish between sets of gear within a playbook such as The Professional, who can choose one 'serious' weapon and two 'normal' weapons.  Currently also includes means of transport if the playbook has them, keyworded as 'transport'. The new seed file will populate these.
* Added `after` block to Gear seeds to ensure it runs after the playbook is created.
* Updated spec files to reflect the new architecture.
* Added Categories to PlaybookGear through table so that we can differentiate when a Playbook has different categories of gear.

### **Notes:**
* You were asking on stream briefly about the [spec file for hunters_improvements](spec/controllers/hunters_improvements_controller_spec.rb) and why I changed it to include `to_s`.   Previously, the test for the hunters_improvements JSON error message would pass because it was set with a matcher that included the exact text it would respond with ('does not match improvement playbook: ').  After adding a name to the factory it would no longer pass, as it was attempting to find 'does not match improvement playbook: ' within an array that contained only one index that read 'does not match improvement playbook: The Unspecified'. I converted that object into a string so that it checked for the same thing (that the array contained the error text) but was not dependant on hard coding in the name of the factory, should we ever decide to change it, have multiple factories, or use Faker.

resolves #96

### CODE REVIEW

**Comment by LeftoverCurry** [comment]
### **Issue:**
Gear was originally set up to belong to an individual playbook, but in reality the same gear could belong to many different playbooks.

### **Resolution:**
* Created a `has_many: :through` ActiveRecord association for each table and set up a through table with the model PlaybookGear.  
* Restructured and built out the Gear seeds to contain a static list of available gear.
* Restructured the currently built out Playbook seeds to reference the gear ID and build the through table.
* Added `after` block to Playbook seeds to ensure it runs after the gear is created.
* Removed `playbook_id` from the Gear table.
* Used the currently existing and unused `description` column in the Gear table to provide data to make selections when a Playbook has multiple sets of gear, such as The Professional, who can choose one 'serious' weapon and two 'normal' weapons.
* Updated spec files to reflect the new architecture.

### **Notes:**
* This will require a `db:reset`, as the Gear references only work if they are in the correct order in the database since they are selected by ID.
resolves #96

File: db/migrate/20200526200555_add_gear_to_playbook.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
```
Comment (ChaelCodes) [inline db/migrate/20200526200555_add_gear_to_playbook.rb:1]: Can you remove this migration from the commit history?

File: db/migrate/20200526205042_remove_playbook_id_from_gears.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
```
Comment (ChaelCodes) [inline db/migrate/20200526205042_remove_playbook_id_from_gears.rb:1]: And then we can remove this one!

File: db/migrate/20200526230159_remove_available_gears_from_playbooks.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class RemoveAvailableGearsFromPlaybooks < ActiveRecord::Migration[6.0]
```
Comment (ChaelCodes) [inline db/migrate/20200526230159_remove_available_gears_from_playbooks.rb:3]: And then we can remove this one!

File: db/seeds/playbooks/crooked.seeds.rb [hunk 4]
```diff
@@ -274,57 +274,21 @@
...
+  # 22 revolver (1-harm close reload small) -1
+  # .38 revolver (2-harm close reload loud) -2
+  # 9mm (2-harm close loud) - 3
+  # Shotgun (3-harm close messy) - 62
+  # Hunting rifle (2-harm far loud) - 46
+  # Big knife (1-harm hand) - 12
+  # Baseball bat (1-harm hand) - 8
+  # Submachinegun (2-harm close reload area) - 73
+  # Assault rifle (3-harm close/far area) - 5
+  [1, 2, 3, 62, 46, 12, 8, 73, 5].each do |gear_id|
```
Comment (ChaelCodes) [inline db/seeds/playbooks/crooked.seeds.rb:288]: This list of ids will be different in each environment, especially production, which already gear loaded. Name and Harm would be a better way to associate this.

File: spec/models/playbook_spec.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
```
Comment (ChaelCodes) [inline spec/models/playbook_spec.rb:1]: Thank you for creating this spec! There was one in minitest that needs to be deleted, not that it has been migrated to RSpec.
Please remove this file:
https://github.com/ChaelCodes/HuntersKeepers/blob/10e66ea18cf29db1e2e06b8a65ee2eb9ca63a967/test/models/playbook_test.rb#L15
Thank you for migrating the playbook spec to RSpec!

File: spec/models/playbook_gear_spec.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+require 'rails_helper'
+
+RSpec.describe PlaybookGear, type: :model do
+  let(:gear) { create :gear }
+  let(:playbook) { create :playbook }
+
+  it 'correctly pulls gear from a playbook' do
+    playbook.gears << gear
+    expect(playbook.gears.ids).to include(gear.id)
```
Comment (ChaelCodes) [inline spec/models/playbook_gear_spec.rb:11]: Rails has a helper you can use here:
```suggestion
    expect(playbook.gear_ids).to include(gear.id)
```
Totally optional, but it's a nice piece of syntactic sugar.

File: spec/factories/playbooks.rb [hunk 7]
```diff
@@ -1,16 +1,8 @@
 # frozen_string_literal: true
 
-# == Schema Information
```
Comment (ChaelCodes) [inline spec/factories/playbooks.rb:3]: Can we restore this schema info?

File: db/seeds/playbooks/professional.seeds.rb [hunk 8]
```diff
@@ -214,78 +214,35 @@
...
+
+  #   Normal weapons (pick two):
+  # .38 revolver (2-harm close reload loud) - 2
+  # 9mm (2-harm close loud) - 3
+  # Hunting rifle (2-harm far loud) - 46
+  # Shotgun (3-harm close messy) - 62
+  # Big knife (1-harm hand) - 12
+
+  # serious or normal can be searched by the description column for each gear
+  [31, 21, 4, 36, 69, 37, 74, 2, 3, 46, 62, 12].each do |gear_id|
```
Comment (ChaelCodes) [inline db/seeds/playbooks/professional.seeds.rb:242]: This list of ids will be different in each environment, especially production, which already has gear loaded. Name and Harm would be a better way to associate this.

File: db/seeds/playbook.seeds.rb [hunk 9]
```diff
@@ -1,136 +1,138 @@
...
+     name: 'The Searcher',
+     description: 'There’s still so much to be discovered and
   explained, even now. Perhaps only one event in a thousand is
-  true weirdness — but I’ll investigate them all to find it.',
-}].each do |playbook|
-  Playbook.find_or_initialize_by(name: playbook[:name]).update!(playbook)
-end
+  true weirdness — but I’ll investigate them all to find it.'
+   }].each do |playbook|
+    Playbook.find_or_create_by(name: playbook[:name]).update!(playbook)
```
Comment (ChaelCodes) [inline db/seeds/playbook.seeds.rb:117]: Why was this changed to `find_or_create_by`? I don't think the playbook is valid on creation, because the description isn't included in the attributes.

File: app/models/playbook.rb [hunk 10]
```diff
@@ -4,17 +4,22 @@
...
+#  description    :string
+#  created_at     :datetime         not null
+#  updated_at     :datetime         not null
 #
 # The class or playbook that the hunter has
 # provides unique abilities to the Hunter
 class Playbook < ApplicationRecord
   has_many :improvements
-  has_many :gears
+  has_many :playbook_gears
```
Comment (ChaelCodes) [inline app/models/playbook.rb:18]: I think our dependent option should be `destroy`.

File: app/controllers/hunters_controller.rb [hunk 11]
```diff
@@ -18,8 +18,10 @@ def index
   # @param :available_improvements [Boolean] Toggles whether or not to include a list of improvements available to
   #   the hunter
   # @see Hunter#available_improvements
+  # Available_gear contains all gear available to the Hunter's current playbook
   def show
     @available_improvements = params[:available_improvements]
+    @available_gear = @hunter.playbook.gears
```
Comment (ChaelCodes) [inline app/controllers/hunters_controller.rb:24]: Where do we use `@available_gear`?

File: db/migrate/20200526200555_add_gear_to_playbook.rb (hunk 1 above)
Comment (LeftoverCurry) [reply db/migrate/20200526200555_add_gear_to_playbook.rb]: Removed in commit 6ef0a4a

File: db/migrate/20200526230159_remove_available_gears_from_playbooks.rb (hunk 3 above)
Comment (LeftoverCurry) [reply db/migrate/20200526230159_remove_available_gears_from_playbooks.rb]: Removed in commit 6ef0a4a

File: spec/models/playbook_spec.rb (hunk 5 above)
Comment (LeftoverCurry) [reply spec/models/playbook_spec.rb:1]: Removed with commit 692f05e

File: spec/models/playbook_gear_spec.rb (hunk 6 above)
Comment (LeftoverCurry) [reply spec/models/playbook_gear_spec.rb]: Poured some sugar on it.  Fixed with commit 3c872f3.

File: spec/factories/playbooks.rb (hunk 7 above)
Comment (LeftoverCurry) [reply spec/factories/playbooks.rb]: There appeared to be several files that needed documentation updates.  I ran the annotate gem and included them all in commit dc1a1f1.  I think it might be worth considering adding a task to the rakefile to automate that process whenever a new migration is run. 

From the gem documentation:
Screen Shot 2020-06-01 at 11 02 51 AM

File: db/seeds/playbooks/crooked.seeds.rb (hunk 4 above)
Comment (LeftoverCurry) [reply db/seeds/playbooks/crooked.seeds.rb:288]: We discussed this elsewhere, but just to have a record of what was decided in case we need to look back over it:
Unfortunately, searching by harm and name won't work, because we have entries that share name and harm.  The solution we came up with was to populate the through table through a reference in gear to what playbooks they belong in.  Initial plan is create a new column referencing the playbook names in the gear table, then rewrite the seed files to populate in that manner.

File: db/seeds/playbooks/professional.seeds.rb (hunk 8 above)
Comment (LeftoverCurry) [reply db/seeds/playbooks/professional.seeds.rb:242]: We discussed this elsewhere, but just to have a record of what was decided in case we need to look back over it:
Unfortunately, searching by harm and name won't work, because we have entries that share name and harm. The solution we came up with was to populate the through table through a reference in gear to what playbooks they belong in. Initial plan is create a new column referencing the playbook names in the gear table, then rewrite the seed files to populate in that manner.

File: db/seeds/playbook.seeds.rb (hunk 9 above)
Comment (LeftoverCurry) [reply db/seeds/playbook.seeds.rb:117]: Reverting and will revisit once new gear seeding is completed.

File: app/controllers/hunters_controller.rb (hunk 11 above)
Comment (LeftoverCurry) [reply app/controllers/hunters_controller.rb]: We do not.  This was added to address the last bullet point on the issue, which ended up not being needed.  Removed in commit c6538bd.

File: app/models/playbook.rb (hunk 10 above)
Comment (LeftoverCurry) [reply app/models/playbook.rb]: Indeed.  Resolved with commit: 977bbfe

File: spec/factories/playbooks.rb (hunk 7 above)
Comment (LeftoverCurry) [reply spec/factories/playbooks.rb]: Added issue #132 to automate documentation.

File: db/seeds/playbooks/crooked.seeds.rb (hunk 4 above)
Comment (LeftoverCurry) [reply db/seeds/playbooks/crooked.seeds.rb]: Addressed in commits d886634 and f886cf9

File: db/seeds/playbooks/professional.seeds.rb (hunk 8 above)
Comment (LeftoverCurry) [reply db/seeds/playbooks/professional.seeds.rb]: Addressed in commits d886634 and f886cf9

**Comment by LeftoverCurry** [comment]
### Correcting Issue With File Seeding Using Index Number

**Ref Commits d886634, f886cf9, cba031e, 4b79985**

* Added a playbooks key to the hash and populated it with which playbooks the gear would belong to using the instance variables declared in the playbooks seed.
* Added a few description keywords I missed on the first run through.
* Broke off the playbooks values from the generation of the gears and used it to populate `Gears.playbooks`.  This prevents the need to create a new column that would duplicate information.
* Removed the `after:` from playbooks and added it to gear, as the seed order had reversed.
* Removed references to gear in the individual playbook's seed files, as we are no longer populating from there.

**Comment by LeftoverCurry** [comment]
### Regarding cba031e
I could not test the app unless I basically adopted the changes from PR #128, so if there are any merge conflicts, let me know and I'll get it cleaned up.

File: app/controllers/hunters_controller.rb [hunk 12]
```diff
@@ -18,6 +18,7 @@ def index
   # @param :available_improvements [Boolean] Toggles whether or not to include a list of improvements available to
   #   the hunter
   # @see Hunter#available_improvements
+  # Available_gear contains all gear available to the Hunter's current playbook
```
Comment (ChaelCodes) [inline app/controllers/hunters_controller.rb:21]: Comment leftover.

File: app/controllers/hunters_controller.rb [hunk 13]
```diff
@@ -32,7 +33,8 @@ def new
   # GET /hunters/1/edit
   #
   # @param id [Integer] id of the Hunter to be edited
-  def edit() end
+  # Available_gear contains all gear available to the Hunter's current playbook
```
Comment (ChaelCodes) [inline app/controllers/hunters_controller.rb:36]: Comment leftover.

File: app/models/gear.rb [hunk 14]
```diff
@@ -9,14 +9,13 @@
 #  description :string
 #  harm        :integer
 #  armor       :integer
-#  playbook_id :bigint
 #  created_at  :datetime         not null
 #  updated_at  :datetime         not null
 #
-# Gear is equipment that the Hunter can use.
```
Comment (ChaelCodes) [inline app/models/gear.rb:16]: This is our top-level comment. I think we still want it. We can get rid of the playbook reference though.

File: app/models/hunter.rb [hunk 15]
```diff
@@ -19,7 +19,6 @@
 #  playbook_id :bigint
 #  user_id     :bigint
 #
-# The character appearing in the mystery
```
Comment (ChaelCodes) [inline app/models/hunter.rb:22]: That's a top-level comment that we want to keep.

File: app/models/playbook.rb [hunk 16]
```diff
@@ -10,11 +10,13 @@
...
 class Playbook < ApplicationRecord
   has_many :improvements
-  has_many :gears
+  has_many :playbook_gears, dependent: :destroy
+  has_many :gears, through: :playbook_gears
   has_many :moves
   has_many :ratings, dependent: :destroy
+
+  validates :name, presence: true
+  validates :description, presence: true
```
Comment (ChaelCodes) [inline app/models/playbook.rb:21]: Like the extra validations! ✨

File: db/seeds/gear.seeds.rb [hunk 17]
```diff
@@ -1,103 +1,101 @@
...
+    { name: 'Throwing Knives', harm: 1, tags: 'close, many', playbooks: [@flake] },
+    { name: 'Thunder Hammer', harm: 3, tags: 'hand, stun, holy', playbooks: [@divine] },
+    { name: 'Truck', tags: '', description: 'professional_transport', playbooks: [@professional] },
+    { name: 'Van', tags: '', description: 'mundane_transport, professional_transport', playbooks: [@mundane, @professional] },
+    { name: "Watchman's Flashlight", harm: 1, tags: 'hand', playbooks: [@flake] },
+    { name: 'Weighted Gloves/Brass Knuckles', harm: 1, tags: 'hand', playbooks: [@flake] }
+  ].each do |gear|
+    tag_list = gear.delete(:tags)
+    playbooks = gear.delete(:playbooks)
+    g = Gear.create!(gear)
```
Comment (ChaelCodes) [inline db/seeds/gear.seeds.rb:96]: What if the gear already exists? We run our seed files every deploy. I believe this would constantly recreate gear in our database.

File: spec/controllers/hunters_improvements_controller_spec.rb [hunk 18]
```diff
@@ -30,7 +30,7 @@
   # HuntersImprovementsController. Be sure to keep this updated too.
   let(:valid_session) { {} }
 
-  before(:each) do
+  before do
```
Comment (ChaelCodes) [inline spec/controllers/hunters_improvements_controller_spec.rb:33]: Was this a linter fix? If so, in another PR, I'd like to configure the Rubocop rules to recommend the more specific option.

File: db/seeds/gear.seeds.rb [hunk 19]
```diff
@@ -1,103 +1,101 @@
...
-   }].each do |gear|
-    tag_list = gear.delete(:tag_list)
-    g = Gear.find_or_create_by!(gear)
+  [
+    { name: '.22 Revolver', harm: 1, tags: 'close, reload, small', playbooks: [@crooked] },
+    { name: '.38 Revolver', harm: 2, tags: 'close, reload, loud', description: 'normal, modern, practical', playbooks: [@crooked, @flake, @initiate, @monstrous, @professional, @spooky, @wronged] },
+    { name: '9mm', harm: 2, tags: 'close, loud', description: 'normal, modern, practical', playbooks: [@crooked, @flake, @initiate, @monstrous, @professional, @spooky, @wronged] },
+    { name: 'Assault Rifle', harm: 3, tags: 'far, area, loud, reload', description: 'serious', playbooks: [@professional] },
+    { name: 'Assault Rifle', harm: 3, tags: 'close/far, area',  playbooks: [@crooked] },
+    { name: 'Assault Rifle', harm: 3, tags: 'close, area, loud, reload', description: 'signature', playbooks: [@wronged] },
```
Comment (ChaelCodes) [inline db/seeds/gear.seeds.rb:10]: What are these descriptions for?

File: db/seeds/gear.seeds.rb [hunk 20]
```diff
@@ -1,103 +1,101 @@
...
+    { name: 'Axe', harm: 2, tags: 'hand, messy', playbooks: [@initiate], description: 'old-fashioned' },
+    { name: 'Baseball Bat', harm: 1, tags: 'hand', playbooks: [@crooked] },
+    { name: 'Baseball Bat', harm: 2, tags: 'hand, innocuous, messy', playbooks: [@mundane] },
+    { name: 'Bicycle', tags: '', playbooks: [@mundane], description: 'mundane_transport' },
```
Comment (ChaelCodes) [inline db/seeds/gear.seeds.rb:14]: This one has underscores???

File: app/controllers/hunters_controller.rb (hunk 12 above)
Comment (LeftoverCurry) [reply app/controllers/hunters_controller.rb:21]: Fixed with commit d7d6308.

File: app/controllers/hunters_controller.rb (hunk 13 above)
Comment (LeftoverCurry) [reply app/controllers/hunters_controller.rb:36]: Fixed with commit d7d6308.

File: spec/controllers/hunters_improvements_controller_spec.rb (hunk 18 above)
Comment (LeftoverCurry) [reply spec/controllers/hunters_improvements_controller_spec.rb:33]: It was a local issue that I have fixed.

Comment (LeftoverCurry) [reply spec/controllers/hunters_improvements_controller_spec.rb]: Resolved with commit d1c055e.

File: db/seeds/gear.seeds.rb (hunk 17 above)
Comment (LeftoverCurry) [reply db/seeds/gear.seeds.rb]: Corrected to `find_or_initialize_by` since the save is called later in the code.  Resolved in commit 3501c33.

File: app/models/gear.rb (hunk 14 above)
Comment (LeftoverCurry) [reply app/models/gear.rb:16]: Resolved with commit 817d7cc.  Tested with the annotate gem and top level documentation was not removed.

File: app/models/hunter.rb (hunk 15 above)
Comment (LeftoverCurry) [reply app/models/hunter.rb:22]: Resolved with commit 817d7cc. Tested with the annotate gem and top level documentation was not removed.

File: db/seeds/gear.seeds.rb (hunk 19 above)
Comment (LeftoverCurry) [reply db/seeds/gear.seeds.rb]: Resolved with commit b8fcd8c.

File: db/seeds/gear.seeds.rb (hunk 20 above)
Comment (LeftoverCurry) [reply db/seeds/gear.seeds.rb]: Resolved with commit b8fcd8c

File: app/controllers/hunters_controller.rb (hunk 12 above)
Comment (ChaelCodes) [reply app/controllers/hunters_controller.rb:21]: I'm still seeing this in the final file chantges.

File: app/controllers/hunters_controller.rb (hunk 13 above)
Comment (ChaelCodes) [reply app/controllers/hunters_controller.rb:36]: Same here, I'm still seeing this in the final file changes.

File: app/controllers/hunters_controller.rb (hunk 12 above)
Comment (LeftoverCurry) [reply app/controllers/hunters_controller.rb]: should be fixed as of 59ce133.

File: app/controllers/hunters_controller.rb (hunk 13 above)
Comment (LeftoverCurry) [reply app/controllers/hunters_controller.rb]: Should be fixed as of 59ce133.

File: db/seeds/gear.seeds.rb [hunk 21]
```diff
@@ -1,103 +1,103 @@
...
+    { name: 'Specialist Weapons (Chosen Foe)', harm: 4, tags: "Keeper's Discretion", keywords: 'signature', playbooks: [@wronged] },
+    { name: 'Submachine Gun', harm: 2, tags: 'close, reload, area', playbooks: [@crooked] },
+    { name: 'Submachine Gun', harm: 3, tags: 'close, area, loud, reload', keywords: 'serious', playbooks: [@professional] },
+    { name: 'Sword', harm: 2, tags: 'hand, messy', playbooks: [@initiate, @monstrous], keywords: 'old-fashioned' },
+    { name: 'Throwing Knives', harm: 1, tags: 'close, many', playbooks: [@flake] },
+    { name: 'Thunder Hammer', harm: 3, tags: 'hand, stun, holy', playbooks: [@divine] },
+    { name: 'Truck', tags: '', keywords: 'transport', playbooks: [@professional] },
+    { name: 'Van', tags: '', keywords: 'transport', playbooks: [@mundane, @professional] },
+    { name: "Watchman's Flashlight", harm: 1, tags: 'hand', playbooks: [@flake] },
+    { name: 'Weighted Gloves/Brass Knuckles', harm: 1, tags: 'hand', playbooks: [@flake] }
```
[the remark this answers is not in the document]
Comment (LeftoverCurry) [reply db/seeds/gear.seeds.rb:92]: So, the issue here is that the playbooks don't know what categories they want, so it's always going to pick whatever category comes first (or last, if you're overwriting).  So if you're populating, say, a 9mm, that has multiple categories it can fit into (Normal, Modern, and Practical, in this case), all PlaybookGear entries for 9mm will categorized as normal, even if the referenced playbook does not have that category.  My fix for this would be to include an acts_as_taggable on Playbook tagging the categories that the playbook itself includes and then using that in the seed file to distinguish which categories it's looking for.

File: app/models/playbook.rb [hunk 22]
```diff
@@ -10,11 +12,14 @@
...
+  acts_as_taggable_on :categories
```
Comment (LeftoverCurry) [inline app/models/playbook.rb:24]: Added a category tag to playbooks here so that the playbooks knew which categories to seed with.

File: app/models/playbook_gear.rb [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+#  id          :bigint           not null, primary key
+#  playbook_id :bigint           not null
+#  gear_id     :bigint           not null
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+# Through table model for Playbook <=> Gear
+class PlaybookGear < ApplicationRecord
+  belongs_to :playbook
+  belongs_to :gear
```
[the remark this answers is not in the document]
Comment (LeftoverCurry) [reply app/models/playbook_gear.rb:17]: Added migration.

File: db/seeds/gear.seeds.rb (hunk 21 above)
[the remark this answers is not in the document]
Comment (LeftoverCurry) [reply db/seeds/gear.seeds.rb:94]: Added below.

File: db/seeds/playbook.seeds.rb [hunk 24]
```diff
@@ -13,7 +14,8 @@
    description: 'Yeah, I’ve been around the block. A bit of this, a
 bit of that. When I came across the secret underworld of monsters
 and magic… well… it wasn’t so different from the underworld I
-already knew. It was easy to find an angle, just like before.'
+already knew. It was easy to find an angle, just like before.',
+   categories: 'transport'
```
Comment (LeftoverCurry) [inline db/seeds/playbook.seeds.rb:18]: Added categories here.  I do feel like they would fit better in the individual playbooks, but I did it here as a work around solution until the individual playbooks are built out.

File: db/seeds/playbook.seeds.rb [hunk 25]
```diff
@@ -113,7 +120,12 @@
   explained, even now. Perhaps only one event in a thousand is
   true weirdness — but I’ll investigate them all to find it.'
  }].each do |playbook|
-  Playbook.find_or_initialize_by(name: playbook[:name]).update!(playbook)
+  pb = Playbook.find_or_initialize_by(name: playbook[:name])
+  if playbook.key?(:categories)
+    categories = playbook.delete(:categories)
+    pb.category_list = categories
+  end
+  pb.update(playbook)
```
Comment (LeftoverCurry) [inline db/seeds/playbook.seeds.rb:128]: See comment above.  We needed a way for the playbooks to know which categories they had so that if a gear could be in multiple categories it wouldn't just pick whatever category was first.  I built out tags to indicate them and saved them in the category_list for the playbook.

File: spec/controllers/gears_controller_spec.rb [hunk 26]
```diff
@@ -7,7 +7,6 @@
   # adjust the attributes here as well.
   let(:valid_attributes) {
     {
```
Comment (LeftoverCurry) [inline spec/controllers/gears_controller_spec.rb:9]: We no longer need this attribute and it causes the tests to fail.

File: spec/factories/playbook_gears.rb [hunk 27]
```diff
@@ -2,18 +2,16 @@
...
 #
-require 'test_helper'
-
-class PlaybookTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+FactoryBot.define do
+  factory :playbook_gear do
+  end
```
Comment (LeftoverCurry) [inline spec/factories/playbook_gears.rb:16]: Removed Minitest and migrated to RSpec.

File: spec/models/playbook_gear_spec.rb [hunk 28]
```diff
@@ -0,0 +1,29 @@
...
+
+  it 'correctly pulls gear from a playbook' do
+    playbook.gears << gear
+    expect(playbook.gear_ids).to include(gear.id)
+  end
+
+  it 'correctly pulls playbooks from gear' do
+    playbook.gears << gear
+    expect(gear.playbook_ids).to include(playbook.id)
+  end
```
Comment (LeftoverCurry) [inline spec/models/playbook_gear_spec.rb:28]: Added a model test for PlaybookGear.

File: .gitignore [hunk 29]
```diff
@@ -37,3 +37,7 @@ yarn-debug.log*
 # Ignore Compiled yardoc
 /.yardoc
 /doc
+
+# Ignore VSCode debugger config
+/.vscode
+/bin/rspec
```
Comment (LeftoverCurry) [inline .gitignore:43]: Allows a local user to use VSCode's debugger.

File: app/controllers/gears_controller.rb [hunk 30]
```diff
@@ -72,6 +72,6 @@ def set_gear
 
   # Never trust parameters from the scary internet, only allow the white list through.
   def gear_params
-    params.require(:gear).permit(:name, :description, :harm, :armor, :playbook_id, :tag_list)
+    params.require(:gear).permit(:name, :description, :harm, :armor, :tag_list)
```
Comment (LeftoverCurry) [inline app/controllers/gears_controller.rb:75]: The Playbook relationship is now handled under PlaybookGear.

File: app/models/gear.rb [hunk 31]
```diff
@@ -1,5 +1,13 @@
 # frozen_string_literal: true
 
+# Gear is equipment that the Hunter can use.
+# Gear includes tags (through the acts_as_taggable_on gem), which are used
+# by the Keeper and Hunter and are visible to users and keywords, which are used
+# to distinguish different sets of gear that might
+# have different numbers of selectable gear, such as with The Flake, who can
+# have one 'normal' weapon and two 'hidden' weapons.  Keywords can also be used
+# to reference modes of transport using the keyword 'transport'.
+
```
Comment (LeftoverCurry) [inline app/models/gear.rb:10]: All user generated top level comments have been moved above the documentation generated by the annotate gem, as it deletes anything between it and the class.

File: app/views/gears/index.html.erb [hunk 32]
```diff
@@ -23,7 +23,6 @@
         <td><%= gear.harm %></td>
         <td><%= gear.armor %></td>
         <td><%= gear.tag_list %></td>
-        <td><%= link_to_playbook(gear) %></td>
         <td><%= link_to 'Show', gear %></td>
```
Comment (LeftoverCurry) [inline app/views/gears/index.html.erb:26]: Gear is no longer associated with an individual playbook.

File: db/migrate/20200526160952_create_playbook_gears.rb [hunk 33]
```diff
@@ -0,0 +1,10 @@
+class CreatePlaybookGears < ActiveRecord::Migration[6.0]
+  def change
+    create_table :playbook_gears do |t|
+      t.references :playbook, null: false, foreign_key: true
+      t.references :gear, null: false, foreign_key: true
+
```
[the remark this answers is not in the document]
Comment (LeftoverCurry) [reply db/migrate/20200526160952_create_playbook_gears.rb:6]: I added a migration below to add it.

File: db/seeds/playbook.seeds.rb [hunk 34]
```diff
@@ -134,3 +146,4 @@
 @pararomantic = Playbook.find_by name: 'The Pararomantic'
 @searcher = Playbook.find_by name: 'The Searcher'
 @initiate = Playbook.find_by name: 'The Initiate'
+@playbooks = Playbook.all
```
Comment (LeftoverCurry) [inline db/seeds/playbook.seeds.rb:149]: Added a general helper here for the Playbook Collection.

File: spec/factories/playbooks.rb [hunk 35]
```diff
@@ -12,5 +12,7 @@
 #
 FactoryBot.define do
   factory :playbook do
+    name { 'The Unspecified' }
+    description { 'A mysterious wonderer' }
```
Comment (LeftoverCurry) [inline spec/factories/playbooks.rb:16]: Needed a name to check for in one of the tests.

**Comment by ChaelCodes** [comment]
You built a genuinely good PR here. Looking at the differences between gear on the different playbooks though, I think it'd be better to maintain a 1-to-many instead of many-to-many. Sorry!
